heareval/tasks/speech_commands.py

```diff
@@ -2,38 +2,183 @@
-
-import heareval.tasks.config.speech_commands as config
-import heareval.tasks.util.audio as audio_util
-import heareval.tasks.util.luigi as luigi_util
-from heareval.tasks.util.luigi import DownloadCorpus, WorkTask, ExtractArchive
-
+import pandas as pd
+from heareval.tasks.util.luigi import (
+    PROCESSMETADATACOLS,
+    DownloadCorpus,
+    ExtractArchive,
+    FinalizeCorpus,
+    MetadataVocabulary,
+    MonoWavTrimCorpus,
+    ResampleSubCorpus,
+    SplitTrainTestCorpus,
+    SplitTrainTestMetadata,
+    SubsampleCorpus,
+    WorkTask,
+    ensure_dir,
+    filename_to_int_hash,
+    slugify_file_name,
+    which_set,
+)
 
 # Set the task name for all WorkTasks
 WorkTask.task_name = config.TASKNAME
 
 
+class ExtractArchive(ExtractArchive):
+    def requires(self):
+        return {
+            "download": DownloadCorpus(url=config.DOWNLOAD_URL, outfile="corpus.tar.gz")
+        }
+
+
+class ConfigureProcessMetaData(WorkTask):
+    """
+    This config is data dependent and has to be set for each data
+    """
+
+    def requires(self):
+        return ExtractArchive(infile="corpus.tar.gz")
+
+    def run(self):
+        # Get relative path of the audio files
+        # This file can also be built with the metadata file for the dataset
+        # in case the metadata is provided
+        process_metadata = pd.DataFrame(
+            glob(os.path.join(self.requires().workdir, "[!_]*/*.wav")),
+            columns=["relpath"],
+        )
+
+        # Get unique slug for each file
+        # In this case the base name is not excluded since same person have made dataset
+        # for different category
+        # This is something which is data dependent and should be handled here
+        process_metadata["slug"] = (
+            process_metadata["relpath"]
+            # Dont exclude the base name. Apply sluggify on all
+            .apply(partial(os.path.relpath, start=self.requires().workdir)).apply(
+                slugify_file_name
+            )
+        )
+
+        # Hash the field id rather than the full path.
+        # This hashing is specific to the dataset and should be done here
+        # In this case we take the slug and remove the -nohash- as described
+        # This nohash removal allows for the speech of similar person to be in the
+        # same dataset. Such type of data specific requirements might be there.
+        # in the readme of google speech commands. we want to keep similar people
+        # in the same group - test or train or val
+        process_metadata["filename_hash"] = (
+            process_metadata["slug"]
+            .apply(lambda relpath: re.sub(r"-nohash-.*$", "", relpath))
+            .apply(filename_to_int_hash)
+        )
+
+        # Get the data partition either from the input file or by doing the hash
+        process_metadata["partition"] = process_metadata["filename_hash"].apply(
+            partial(which_set, validation_percentage=0.0, testing_percentage=10.0)
+        )
```

Comment:
The `filename_hash` and the `partition` are made as part of the `Configure Process Metadata`. The reason to do it here is that, a new dataset might have another test train split strategy and that we can capture that in the `Configure Process Metadata`. I though of doing the sampling of the configuration data here as well so that we can stratify it based on the label. This was we will have the partition name, label for the all the subsampled files and in the next step we can go ahead with the next steps.